### PR TITLE
Update terms of use version mapping

### PR DESCRIPTION
**Pull request by rileyanderson**
## Summary
Update terms of use version mapping and add `legal_display_version` field

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#72039

## What areas of the site does it impact?
terms of use

### CODE REVIEW

File: lib/map/sign_up/configuration.rb [hunk 1]
```diff
@@ -44,11 +44,15 @@ def provisioning_acceptable_status
 
       def agreements_version_mapping
         {
-          'v1' => 2,
-          'v2' => 3
```
Comment (bosawt) [inline lib/map/sign_up/configuration.rb:48]: we can prob remove the 'v2' here

File: lib/map/sign_up/service.rb [hunk 2]
```diff
@@ -59,7 +59,8 @@ def agreements_body(icn, signature_name, version)
           responseDate: Time.zone.now,
           icn:,
           signatureName: signature_name,
-          version: config.agreements_version_mapping[version]
+          version: config.agreements_version_mapping[version],
+          legalDisplayVersion: config.legal_display_version
```
Comment (bosawt) [inline lib/map/sign_up/service.rb:63]: did this not change specs?

Comment (rileyanderson) [reply lib/map/sign_up/service.rb:63]: I updated the cassette to match on the request body

**Review by bosawt — APPROVED** [review]
looks good

**Pull request merged**